pkgs/development/interpreters/python/hooks/python-namespaces-hook.sh

```diff
@@ -17,16 +17,19 @@ pythonNamespacesHook() {
         for pathSegment in ${pathSegments[@]}; do
             constructedPath=${constructedPath}/${pathSegment}
             pathToRemove=${constructedPath}/__init__.py
-            pycachePath=${constructedPath}/__pycache__/__init__*
+            pycachePath=${constructedPath}/__pycache__/
 
             if [ -f "$pathToRemove" ]; then
                 echo "Removing $pathToRemove"
                 rm "$pathToRemove"
             fi
 
-            if [ -f "$pycachePath" ]; then
-                echo "Removing $pycachePath"
-                rm "$pycachePath"
+            if [ -d "$pycachePath" ]; then
+                # can be interpreter specific. E.g. __init__.cpython-38.pyc
+                for f in $(@findutils@/bin/find $pycachePath -name '__init__*'); do
+                    echo "Removing $f"
+                    rm "$f"
+                done
```

Comment:
Perhaps

```suggestion
                $(@findutils@/bin/find $pycachePath -name '__init__*' -print0 | xargs -0 rm -v

```

Another option would be to use
```
                    -exec echo Removing '{}' ';' \
                    -exec rm '{}' ';'
```

The only way I can think of that it would matter is if someone has spaces (or other whitespace) in their `NIX_STORE_DIR` which I suspect we wouldn't handle correctly in general.

Reply:
changing the NIX_STORE_DIR would mean having to rebuild everything... IIRC, nix checks to see if the exact store path exists.

Reply:
Yes, though unprivileged users may not have much of a choice. Though anyone with a space in the name of their home directory must have a real BOFH. :laughing: 

The other way this could break is if somehow something other than a valid python module ended up under a __pycache__ directory, which also seems pretty unlikely.

Anyway, it seems like a very minor issue at worst. The only reason I brought it up is that the surrounding code is careful to properly quote filenames. :shrug:

Reply:
added, thanks!